compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1222,11 +1208,6 @@ pub trait WithConstness: Sized {
         ConstnessAnd { constness, value: self }
     }
 
-    #[inline]
-    fn with_const(self) -> ConstnessAnd<Self> {
-        self.with_constness(Constness::Const)
-    }
```

Comment:
Please keep this around, we'd need to re-add it in the future when implementing more `const Trait` things